### ISSUE TITLE
Seriously OnePlus ? When do we get src we can build?

### ISSUE THREAD

**Issue by vczilla** [OnePlusOSS/android_kernel_msm-5.10_oneplus_sm8450#4]
Lo and behold the Android 13 kernel source still doesn't compile.
Several different reasons for it: the building system is picking the wrong .config fragments or 
rather the incomplete ones because some files that are referred to are not there.
(build.config.msm.wapio.oplus ( note that waipio was also misspelt) , modules.list.oplus being also conscipcuously absent ). Even what's there wouldn't allow to build the kernel since you end up with a defconfig that lacks both CONFIG_QCOM_SMEM and  CONFIG_OPLUS_FEATURE_OPROJECT that will prevent the linking phase to work.

Also talking about project_info the same annoying bugs are still there when it's trying to compile against the QCOM variant: it's using a struct file_operations for the project_info_fops variable instead of using a struct proc_ops and it's also using boot_command_line instead of saved _command_line.

Plus obviously the stock kernel is actually using gki/project_info.c and not qcom/project_info.c even though the last one gets picked up because of a config setting  CONFIG_OPLUS_KERNEL_QCOM
How difficult would it be to release the source to a fully working kernel ?
Answer not whatsoever since Asus was able to do it .

**Comment by EviraKernel** [comment]
idk why oneplus stop listen dev community we wait working source wake up oneplus

**Comment by Proxy13** [comment]
@vczilla did u try 13.0 branch ? it seem same :S

**Comment by Proxy13** [comment]
@OnePlusOSSAdmin

**Comment by SaeedDev94** [comment]
@EviraKernel Probably because it's not OnePlus anymore, it's OPPO

**Comment by Proxy13** [comment]
@vczilla maybe u can patch it using this repe.

https://github.com/severecold/oplus_oss_sm8450_kernel

**Comment by vczilla** [comment]
Well I patched by not using it and using Realme's kernel and vendor src code for the gt2 pro.

( Like you did with Oppo's vendor stuff I'm not mistaken).

Actually I ended up using qcom"s vanilla kernel for the waipio soc ( which was 5.10.136 ) and I slowly reintroduced Realme's vendor stuff that wasn't in qcom 's kernel .

End result for now; it boots into recovery ( stock and twrp ) ( with touchscreen working ) but not into system ( and I tried both OOS and a GSI ).

I reckon I missed something obvious regarding vendor_dlkm but I haven't found what yet .

**Comment by p1ngx** [comment]
@OnePlusOSSAdmin, once upon a time, everyone bought your company's phones because you listened to your company's customers. Your phones could be customized, thanks to good kernel sources, but now the situation is completely different. Many people who have already fallen in love with your phones will leave you soon. Personally, I am selling my OnePlus 10 Pro and will never buy your company's phone again. I advise others to do this as well.

**Comment by SaeedDev94** [comment]
> @OnePlusOSSAdmin, once upon a time, everyone bought your company's phones because you listened to your company's customers. Your phones could be customized, thanks to good kernel sources, but now the situation is completely different. Many people who have already fallen in love with your phones will leave you soon. Personally, I am selling my OnePlus 10 Pro and will never buy your company's phone again. I advise others to do this as well.

Same here, No more BugPlus.

**Comment by DeeparthGupta** [comment]
Is this a GPL violation? It should be reported if so.

**Comment by Proxy13** [comment]
@vczilla Can you share your improvements and patches with us?

which tag or branch based qcom"s vanilla kernel for the waipio soc used ?

**Comment by OnePlusOSSAdmin** [comment]
Following missing config file would be released. If any more file or information is needed, we'll provide.
abi_gki_aarch64_oplus_internal
build.config.msm.wapio.oplus
modules.ext.oplus
modules.list.oplus
waipio_oplus.config

**Comment by p1ngx** [comment]
> Following missing config file would be released. If any more file or information is needed, we'll provide. abi_gki_aarch64_oplus_internal build.config.msm.wapio.oplus modules.ext.oplus modules.list.oplus waipio_oplus.config

oplus_project.h

**Comment by OnePlusOSSAdmin** [comment]
Code has been updated. 
https://github.com/OnePlusOSS/android_kernel_modules_and_devicetree_oneplus_sm8450/tree/oneplus/sm8450_t_13.0_10pro
@stonmick oplus_project.h is located in vendor/oplus/kernel/boot/include.

**Comment by yaymalaga** [comment]
@OnePlusOSSAdmin Could you also check the 10T kernel missing parts? https://github.com/OnePlusOSS/android_kernel_modules_and_devicetree_oneplus_sm8475/issues/1

**Comment by OnePlusOSSAdmin** [comment]
@yaymalaga Sure. It'll be updated.

**Comment by x1906875173** [comment]
@OnePlusOSSAdmin Will msmtool be released to users in the future?

**Comment by p1ngx** [comment]
> Code has been updated. https://github.com/OnePlusOSS/android_kernel_modules_and_devicetree_oneplus_sm8450/tree/oneplus/sm8450_t_13.0_10pro @stonmick oplus_project.h is located in vendor/oplus/kernel/boot/include.

Thanks a lot

**Comment by DeeparthGupta** [comment]
> @OnePlusOSSAdmin Will msmtool be released to users in the future?

+1
Though I don't think this is the right place to ask this question. At the same time, OP doesn't seem to want to respond to questions like this on other channels.

**Comment by OnePlusOSSAdmin** [comment]
@x1906875173 @DeeparthGupta This tool is not managed by OSS team. This tool is for internal use only and there's no plan to release it to users. If you need help, you could contact customer support.

**Comment by Proxy13** [comment]
is anyone boot successfully the kernel ? @vczilla

**Comment by Proxy13** [comment]
@OnePlusOSSAdmin we (me and @EviraKernel ) can finally compile kernel source after months .But cant boot kernel.Phone goes black screen . are u sure source is complete ?

**Comment by Proxy13** [comment]
if someone try to compile our last point is -> 
https://github.com/negroni-development/OOS-sm8450_t_13.0_10pro

**Comment by EviraKernel** [comment]
@OnePlusOSSAdmin can relase source with gki repo can syncronize and compile how Google do with pixel 7/7 pro?  Source compile go all ok in build with latest source relase but go fail where try to boot same with GT2pro for realme source relase it

**Comment by JimmyMultani** [comment]
Hey @OnePlusOSSAdmin, it would be great if we could get a response about the kernel source not booting.

**Comment by EviraKernel** [comment]
Oneplus and realme relase not booting source idk why relase source when not boot this is violation of gpl so relase working build or relase a working method to build kernel

**Comment by OnePlusOSSAdmin** [comment]
Could you provide more detailed information. After code update, we think code is complete. It would be helpful if we could know wich specific files or configs needed. Also you could send detailed information to oss@oneplus.com

**Comment by x1906875173** [comment]
> Could you provide more detailed information. After code update, we think code is complete. It would be helpful if we could know wich specific files or configs needed. Also you could send detailed information to [oss@oneplus.com](mailto:oss@oneplus.com)

The problem is that we currently have no way of knowing where the problem is

**Comment by EviraKernel** [comment]
oss@oneplus.com actually we not idk why not boot if see on top we relase link of source gki with source updated relase it for build we have all need for boot but not boot phone go on fast boot so or relase a working method  with linked clang gcc and command use to build how oneplus do or we after some months not have working kernel source becouse we build all but phone continue not boot

**Comment by Proxy13** [comment]
@OnePlusOSSAdmin u can't be serious.I think the test is not too hard.Why not try compile released source and try to flash any device ? I think u are just kidding us ...

**Comment by Proxy13** [comment]
@OnePlusOSSAdmin pls update 13.1 .we need tested source .

**Comment by x1906875173** [comment]
@OnePlusOSSAdmin In the past, OnePlus actively listened to our opinions, recognized our problems, and supported the development of developers. Since OnePlus entered OPPO, everything has become a thing of the past. No one has successfully developed your new devices until now. You are preventing developers from developing. You must have tampered with the open source parts, and you are not willing to release the files that are missing in open source. You just want to ensure that the compilation is successful, and you are not responsible for the follow-up problems. The problem lies with you. , and are unwilling to listen to other people's opinions. Now you are only willing to release new devices and no longer support development. You have already restricted msmtool. There are always people who can test, but you are unwilling to continue to support development. It doesn't matter if you don't care, you have already violated your initial wishes.

**Comment by DeeparthGupta** [comment]
Msmtools was always internal and we got access to it though leaks. Thoug it's really just a modified version of a tool released by Qualcomm for public access. Regardless non-booting kernal sources is a GPL violation.

**Comment by W-M-D** [comment]
> @OnePlusOSSAdmin In the past, OnePlus actively listened to our opinions, recognized our problems, and supported the development of developers. Since OnePlus entered OPPO, everything has become a thing of the past. No one has successfully developed your new devices until now. You are preventing developers from developing. You must have tampered with the open source parts, and you are not willing to release the files that are missing in open source. You just want to ensure that the compilation is successful, and you are not responsible for the follow-up problems. The problem lies with you. , and are unwilling to listen to other people's opinions. Now you are only willing to release new devices and no longer support development. You have already restricted msmtool. There are always people who can test, but you are unwilling to continue to support development. It doesn't matter if you don't care, you have already violated your initial wishes.

It's actually funny because the older phones are actually worth more since the OPPO merger.

**Comment by W-M-D** [comment]
> @EviraKernel Probably because it's not OnePlus anymore, it's OPPO

yeah, i knew i should have stayed away from this state owned spyware garbage

**Comment by Machad3x** [comment]
@OnePlusOSSAdmin How to correctly build this kernel? We're not able to boot, you guys even tested this source?

That's a GPL violation, a kernel source need to be complete and functional.

**Comment by Proxy13** [comment]
@Machad3x  we did it. finally can compile kernel succesfully but device did not boot when we try. we got all kernel modules and img files. if u wanna try compile check older comments u can found complete source .

**Comment by W-M-D** [comment]
@Proxy13 yeah, without edl testing kernels is too risky. i'm returning

**Comment by OnePlusOSSAdmin** [comment]
We'll provide a guide on how to compile kernel and flash it to device. It's expected to be released in August.

**Comment by x1906875173** [comment]
> We'll provide a guide on how to compile kernel and flash it to device. It's expected to be released in August.

Have you ever booted the device with your own distributed kernel sources?

**Comment by EviraKernel** [comment]
Guide for make kernel 🤣🤣🤣 you tested source relase it and boot on phone? For kernel build major error is on build  superimage stop compile  and have error

**Comment by pjgowtham** [comment]
If you could include modules.list for both vendor_boot and vendor_boot:recovery and vendor_dlkm, it would make sense considering system boots with OSS kernel while recovery does not. It could be an issue with source code or modules themselves.

**Comment by OnePlusOSSAdmin** [comment]
@x1906875173 Yes. We would test code before we release it to github. But internal we use the full system build, not just kernel. 
@EviraKernel For kernel build error, you could provide error log then we could check what's wrong. Maybe it's because of incorrect config, or other parameters that not included in the code which should be.
We would improve developers' experience by releasing complete compile and install guide.

**Comment by spectrum70** [comment]
Something is changed, sure, but i don't think a bad atmosphere here will bring anything good.

I am not a big expert of android, but, is it possible to enable the low level console and see where the kernel hangs ?

@OnePlusOSSAdmin is there any update ? August is near :)

**Comment by OnePlusOSSAdmin** [comment]
@spectrum70 There's still some work to do before it could be released to developers.

**Comment by pjgowtham** [comment]
Doing this fixes the issue of recovery not booting with OSS kernel modules. Maybe some touchpanel codes are incomplete. Kindly look into it when you update the source.

https://github.com/udon-development/android_device_oneplus_sm8450-common/commit/f036a8cafaf610c1a2d33d218de4c2840caeb2dd

**Comment by Machad3x** [comment]
@OnePlusOSSAdmin Any update about the guide or when will be released? We already in mid August.

**Comment by OnePlusOSSAdmin** [comment]
We could release part of work this week. Here‘s our plan.
1, release build script
2, release tool chain or download link.
3, release manifest

**Comment by KawaiiHachimi** [comment]
hope it's also applicable to SM8475 devices

**Comment by EviraKernel** [comment]
@OnePlusOSSAdmin we wait no news?

**Comment by AngelaCooljx** [comment]
Happy new week!😘

**Comment by p1ngx** [comment]
@OnePlusOSSAdmin, we are still waiting 🫠

**Comment by SaeedDev94** [comment]
It seems that week will never come 😏

**Comment by Machad3x** [comment]
@OnePlusOSSAdmin We're still waiting! 😐

**Comment by AngelaCooljx** [comment]
Happy new week.🙄

**Comment by Machad3x** [comment]
@OnePlusOSSAdmin Have a roadmap to realease? Updated source? Waiting Android 14 release? Plz, tell us.

**Comment by AngelaCooljx** [comment]
Happy new week.😏

**Comment by EviraKernel** [comment]
This week when come?

**Comment by AngelaCooljx** [comment]
Happy new week.😥

**Comment by DeeparthGupta** [comment]
We're looking at a GPL violation at this point.

**Comment by Proxy13** [comment]
@OnePlusOSSAdmin ??

**Comment by EviraKernel** [comment]
@OnePlusOSSAdmin this is becouse i hope Never  continue buy oppo/oneplus/realme device after merge oneplus still bad and not listen community

**Comment by SaeedDev94** [comment]
> @OnePlusOSSAdmin this is becouse i hope Never continue buy oppo/oneplus/realme device after merge oneplus still bad and not listen community

No more OnePlus phones for sure 
The fact is the old and gold OnePlus died a lot of time ago ...  
We all fooled by OPPO ...  
OPPO used OnePlus name to sell this piece of shit to us 😑 R.I.P. OnePlus

**Comment by Kikoarl** [comment]
This has been my first and for sure will be the last OPPO phone I'll ever buy. And I'm going to make sure that no one around me buys another phone from this joke of a company.

**Comment by chandu078** [comment]
> We could release part of work this week. Here‘s our plan. 1, release build script 2, release tool chain or download link. 3, release manifest

This is joke?

**Comment by crackedpotato007** [comment]
Its been a year...

**Comment by MandySingh3** [comment]
> We could release part of work this week. Here‘s our plan. 1, release build script 2, release tool chain or download link. 3, release manifest

...............

**Comment by UploadMaster1** [comment]
> We could release part of work this week. Here‘s our plan. 1, release build script 2, release tool chain or download link. 3, release manifest

@OnePlusOSSAdmin now its more than a year i didn't see these anywhere, i think i am blind can you please tag me if this is not anywhere please remove this crap source from here its giving bootloop if you think we are joking let me know i will screen record and build kernel with this shit source and send you boot.img and you can flash on your device let us know if that works or if you think this source works good, from scratch git clone this repo and build kernel and provide us terminal log if you cant do it on your system i will provide remote access to my system just let me know.

**Comment by OnePlusOSSAdmin** [comment]
We've done it for other chipset platform, pelase see https://github.com/OnePlusOSS/kernel_manifest.
For OnePlus 10R, we'll update to latest code for android 14, plus kernel manifest and build command.

**Comment by herokrat** [comment]
Sorry for bothering you. But we want source here not for another model, why is it so difficult to upload a good working source kernel for the community.?

**Comment by OnePlusOSSAdmin** [comment]
https://github.com/OnePlusOSS/kernel_manifest/tree/oneplus/sm8450
Please check this one.

**Comment by AlexDev404** [comment]
I hate seeing developers cry. Now I'm not sure if buying this OnePlus phone online would even be a right decision... or if I would just be feeding the beast. Because this actually breaks my heart </3

**Comment by SaeedDev94** [comment]
OPPO is selling their phones via OnePlus brand!  
Once for all: OnePlus is dead!  
R.I.P.  
I moved to Google (Pixel), currently I have Pixel 8 Pro, because I care about custom ROMs and software more than hardware  
BTW I wish google improves pixel hardware as well ...

**Comment by AlexDev404** [comment]
> OPPO is selling their phones via OnePlus brand! Once for all: OnePlus is dead! R.I.P. I moved to Google (Pixel), currently I have Pixel 8 Pro, because I care about custom ROMs and software more than hardware BTW I wish google improves pixel hardware as well ...

Yeah currently looking for a developer-friendly phone to replace my old Moto G Power... Maybe you or someone else watching this thread may know of any good options...?

Money isn't the issue here tbh as long as it lasts long enough, and has good developer support (custom ROMS w/ minimal to no glitches, nice root features, good developer support... that kind of stuff--since I like building things for the Linux kernel or working on things like that)

My last phone has lasted me 6 years, but unfortunately is dying of battery issues and hardware bottlenecks now. Looking to find a new phone to replace it with more or less the same timespan (or even better if possible).

(P.S. I absolutely hate the MediaTek crap + am Qualcomm-biased)

**Comment by AlexDev404** [comment]
> OPPO is selling their phones via OnePlus brand! Once for all: OnePlus is dead! R.I.P. I moved to Google (Pixel), currently I have Pixel 8 Pro, because I care about custom ROMs and software more than hardware BTW I wish google improves pixel hardware as well ...

Also yeah was looking at XDA and saw this "OnePlus is dead" a lot. Very sad, sad, story to be honest... Almost as if the OPPO merger hurt everyone, since everyone seemed to love OnePlus a lot.

Such a shame to see a once "friend" slowly turn their back on the very people who helped raise it.

**Comment by SaeedDev94** [comment]
> Maybe you or someone else watching this thread may know of any good options...?

Pixel is my choice, but hardware is not good as other flagships (so be aware if you want to go for it)  
Custom ROMs development is not like before, It's limited to a few models/brand!  
There is no guarantee for it, Sadly, the old time has passed!  
So you should check it by yourself ...  
I believe the best place to looking for it is the LineageOS official devices list  
Do not bet on future (custom ROM) development for a new phone!!  
This is what I did for OnePlus 10 Pro (in 2022) and still there is nothing official for it!!  
Make sure the phone already has active development

**Comment by OnePlusOSSAdmin** [comment]
> https://github.com/OnePlusOSS/kernel_manifest/tree/oneplus/sm8450 Please check this one.

The manifest and build scripts have been released, so we're closing this issue. Please open a new one if you encounter future issues.

**Comment by SaeedDev94** [comment]
Fuck you oppo. 
R.I.P. OnePlus
